File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(new Comparator<Integer>() {
            @Override
            public int compare(Integer o1, Integer o2) {
                return o2 - o1;
            }
        });

        for (int num : nums) {
            pq.add(num);
        }
        int count = 0;
        while (count != k) {
            int result = pq.poll();
            count++;

            if (count == k) {
                return result;
            }
        }

        return count;
    }
}

File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        max_heap = []

        for i in nums:
            heapq.heappush(max_heap, -i)
        
        for i in range(1, k):
            heapq.heappop(max_heap)

        return -heapq.heappop(max_heap)
